native/cocos/renderer/gfx-base/GFXDevice.cpp

```diff
@@ -140,5 +143,34 @@ BufferBarrier *Device::getBufferBarrier(const BufferBarrierInfo &info) {
     return _bufferBarriers[info];
 }
 
+void Device::removeSwapchain(Swapchain *swapchain) {
+    if(!_xr || _swapchains.empty()) return;
+    auto it = _swapchains.begin();
+    while (it != _swapchains.end()) {
+        if (*it == swapchain) {
+            it = _swapchains.erase(it);
+        } else {
+            it++;
+        }
+    }
+}
+
+void Device::destroySwapchains() {
+    if(!_xr) return;
```

Comment:
Does Device::destroySwapchains need only to be used for  XR only?
From the function name of view, it should be a common method.

Reply:
Same as above.

Reply:
?? Same as above??
I mean that `Device::destroySwapchains` is a common method that should not add any xr logic in it.

Reply:
Ok, I got it. You meant you will fix them in the next commit.

Reply:
https://github.com/cocos/cocos-engine/pull/12413/commits/4e59530ce716b151a497ac4688c935be67ea2679